tools/wake-format/emitter.cpp

```diff
@@ -366,62 +393,54 @@ wcl::doc Emitter::walk_block(ctx_t ctx, CSTElement node) {
   return std::move(bdr).build();
 }
 
-wcl::doc Emitter::walk_case(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
+wcl::doc Emitter::walk_case(ctx_t ctx, CSTElement node) {
+  assert(node.id() == CST_CASE);
+
+  return formatter()
+      .walk(WALK(walk_node))
+      .consume_wsnl()
+      .space()
+      .walk(CST_GUARD, WALK(walk_guard))
+      .consume_wsnl()
+      .escape([this](wcl::doc_builder& bdr, ctx_t ctx, CSTElement& node) {
+        wcl::doc doc = space().concat(walk_node(ctx.sub(bdr), node));
+        if (fits(bdr, ctx, doc) && !requires_nl(ctx, node)) {
+          bdr.append(doc);
+        } else {
+          ctx_t n_ctx = ctx.nest();
+          bdr.append(newline(n_ctx));
+          bdr.append(walk_node(n_ctx.sub(bdr), node));
+        }
+        node.nextSiblingElement();
```

Comment:
If this is identical to `def` maybe it *should* be factored out into a seperate function like you had it before. I'm happy to leave it separate and dedup later though. require will also have this chunk of code I guess.